### PR TITLE
Replaced cache actualization for api-keys (through Redis) into RabbitMQ message publishing

### PR DESCRIPTION
**Pull request by polonzer**
https://lykkex.atlassian.net/browse/LWDEV-4236

### CODE REVIEW

File: src/Lykke.Service.HftInternalService.Services/ApiKeyService.cs [hunk 1]
```diff
@@ -1,40 +1,39 @@
...
             if (existedApiKey != null)
             {
-                await _distributedCache.RemoveAsync(GetCacheKey(existedApiKey.Id.ToString()));
                 existedApiKey.ValidTill = DateTime.UtcNow;
                 await _apiKeyRepository.Update(existedApiKey);
+                await PublishNotification(existedApiKey);
             }
             var key = new ApiKey { Id = apiKey, ClientId = clientId, WalletId = walletId };
             await _apiKeyRepository.Add(key);
+            await PublishNotification(key);
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.HftInternalService.Services/ApiKeyService.cs:36]: If PublishNotification(key); throws an exception the key stays in _apiKeyRepository is it normal state?

File: src/Lykke.Service.HftInternalService/Modules/ServiceModule.cs [hunk 2]
```diff
@@ -38,31 +35,22 @@ protected override void Load(ContainerBuilder builder)
...
-                InstanceName = _settings.CurrentValue.HighFrequencyTradingService.CacheSettings.ApiKeyCacheInstance
-            });
-            builder.RegisterInstance(apiKeysRedisCache)
-                .As<IDistributedCache>()
-                .Keyed<IDistributedCache>("apiKeys")
-                .SingleInstance();
+            builder.RegisterType<ApiKeyPublisher>()
+                .As<IApiKeyPublisher>()
+                .As<IStartable>()
+                .SingleInstance()
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.HftInternalService/Modules/ServiceModule.cs]: Should also register AsStopable? Is there anyone who can call Stop?

File: src/Lykke.Service.HftInternalService.Services/ApiKeyService.cs (hunk 1 above)
Comment (polonzer) [reply src/Lykke.Service.HftInternalService.Services/ApiKeyService.cs:36]: It is ok to save key no matter of publishing result. But I'll add a try/catch block for PublishNotification() method.

File: src/Lykke.Service.HftInternalService/Modules/ServiceModule.cs (hunk 2 above)
Comment (polonzer) [reply src/Lykke.Service.HftInternalService/Modules/ServiceModule.cs]: Sure.

**Review by MikhailGerasimov — APPROVED** [review]

**Pull request merged**